server/get_teams_data.go

```diff
@@ -126,6 +128,89 @@ func (p *Plugin) GetMSTeamsTeamChannels(teamID, userID string) ([]msteams.Channe
 	return channels, http.StatusOK, nil
 }
 
+func (p *Plugin) LinkChannels(userID, mattermostTeamID, mattermostChannelID, msTeamsTeamID, msTeamsChannelID string) (responseMsg string, statusCode int) {
+	channel, appErr := p.API.GetChannel(mattermostChannelID)
+	if appErr != nil {
+		p.API.LogError("Unable to get the current channel details.", "ChannelID", mattermostChannelID, "Error", appErr.Message)
+		return "Unable to get the current channel details.", http.StatusInternalServerError
+	}
+
+	if channel.Type == model.ChannelTypeDirect || channel.Type == model.ChannelTypeGroup {
+		p.API.LogError("Linking/unlinking a direct or group message is not allowed.", "ChannelType", channel.Type)
+		return "Linking/unlinking a direct or group message is not allowed", http.StatusForbidden
+	}
+
+	if !p.API.HasPermissionToChannel(userID, mattermostChannelID, model.PermissionManageChannelRoles) {
+		p.API.LogError("Unable to link the channel. You have to be a channel admin to link it.", "ChannelID", mattermostChannelID)
+		return "Unable to link the channel. You have to be a channel admin to link it.", http.StatusForbidden
+	}
+
+	if !p.store.CheckEnabledTeamByTeamID(mattermostTeamID) {
+		p.API.LogError("This team is not enabled for MS Teams sync.", "TeamID", mattermostTeamID)
+		return "This team is not enabled for MS Teams sync.", http.StatusForbidden
+	}
+
+	link, err := p.store.GetLinkByChannelID(mattermostChannelID)
+	if err != nil && err.Error() != ErrorNoRowsInResult {
+		p.API.LogError("Error occurred while getting channel link with Mattermost channelID.", "ChannelID", mattermostChannelID, "Error", err.Error())
+		return "Error occurred while getting channel link with Mattermost channelID.", http.StatusInternalServerError
+	}
+	if link != nil {
+		return "A link for this channel already exists. Please unlink the channel before you link again with another channel.", http.StatusBadRequest
+	}
+
+	link, err = p.store.GetLinkByMSTeamsChannelID(msTeamsTeamID, msTeamsChannelID)
+	if err != nil && err.Error() != ErrorNoRowsInResult {
+		p.API.LogError("Error occurred while getting the channel link with MS Teams channel ID", "Error", err.Error())
+		return "Error occurred while getting the channel link with MS Teams channel ID", http.StatusInternalServerError
+	}
+	if link != nil {
+		return "A link for this channel already exists. Please unlink the channel before you link again with another channel.", http.StatusBadRequest
```

Comment:
The error message should be different in this case.
Something like, "The Teams channel that you're trying to link is already linked to another Mattermost channel. Please unlink that channel and try again."